source/extensions/matching/input_matchers/cel_matcher/matcher.cc

```diff
@@ -0,0 +1,50 @@
+#include "source/extensions/matching/input_matchers/cel_matcher/matcher.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Matching {
+namespace InputMatchers {
+namespace CelMatcher {
+
+CelInputMatcher::CelInputMatcher(const CelExpression& input_expr) {
+  expr_builder_ = Extensions::Filters::Common::Expr::createBuilder(nullptr);
+  switch (input_expr.expr_specifier_case()) {
+  case CelExpression::ExprSpecifierCase::kParsedExpr:
+    compiled_expr_ =
+        Filters::Common::Expr::createExpression(*expr_builder_, input_expr.parsed_expr().expr());
+    return;
+  case CelExpression::ExprSpecifierCase::kCheckedExpr:
+    compiled_expr_ =
+        Filters::Common::Expr::createExpression(*expr_builder_, input_expr.checked_expr().expr());
+    return;
+  case CelExpression::ExprSpecifierCase::EXPR_SPECIFIER_NOT_SET:
+    PANIC_DUE_TO_PROTO_UNSET;
+  }
+  PANIC_DUE_TO_CORRUPT_ENUM;
+}
+
+bool CelInputMatcher::match(const MatchingDataType& input) {
+  Protobuf::Arena arena;
+  if (auto* ptr = absl::get_if<std::shared_ptr<::Envoy::Matcher::CustomMatchData>>(&input);
+      ptr != nullptr) {
+    CelMatchData* cel_data = dynamic_cast<CelMatchData*>((*ptr).get());
+    // Compiled expression should not be nullptr at this point because the program should have
+    // encountered a panic in the constructor earlier if any such error cases occurred. CEL matching
+    // data should also not be nullptr since any errors should have been thrown by the CEL library
+    // already.
+    ASSERT(compiled_expr_ != nullptr && cel_data != nullptr);
```

Comment:
It should RELEASE_ASSERT or add nullptr check later on if you want debug ASSERT.

Reply:
I think we want to avoid the assert in data plane except config path. Here is not the config path.

Actually I am on the fence of having this assert. As i described in comment, it should not trigger. But I think it is good to have for debugging in the future. Initially I add envoy_log, but code coverage complains as I cannot hit it.